src/core/reference/include/ngraph/runtime/reference/acos.hpp

```diff
@@ -13,7 +13,7 @@ namespace reference {
 template <typename T>
 void acos(const T* arg, T* out, size_t count) {
     for (size_t i = 0; i < count; i++) {
-        out[i] = std::acos(arg[i]);
+        out[i] = static_cast<T>(std::acos(arg[i]));
```

Comment:
what is right approach for integer types - round or static case? The same question for other trigonometrical functions.
I see that in asoch we use `std::roundl`